### PR TITLE
Add OctoAI LLM endpoint wrapper and embedding endpoint wrapper

### PR DESCRIPTION
**Pull request by Bassem-Yacoube**
This pull request introduces support for [OctoAI's Compute Service](https://octoai.cloud/) embedding endpoints and large language model endpoints into the LangChain project. The changes encompass additions to various files to accommodate OctoAI's API integration.

**Details of the changes:**

**langchain/__init__.py** and **langchain/llms/__init__.py:** Updates to these initialization files allow OctoAI's services to be utilized in the LangChain's workflow. The changes add 'OctoAIEndpoint' as an additional language model option.

**langchain/embeddings/octoai_embeddings.py:** This new file creates a wrapper for OctoAI embedding endpoints. This addition allows users to compute embeddings of documents and queries using OctoAI's service.

**langchain/llms/octoai_endpoint.py**: This file introduces a new class, 'OctoAIEndpoint', which serves as a wrapper for OctoAI's large language model endpoints. It includes methods for interacting with OctoAI's API and getting responses from it.

**tests/integration_tests/embeddings/test_octoai_embeddings.py**: This file introduces tests to ensure the correct functioning of the OctoAI embedding endpoint using the Instructor-Large model endpoint. It includes tests for document and query embedding functionalities.

**tests/integration_tests/llms/test_octoai_endpoint.py**: This file includes tests for OctoAI's LLM wrapper for MPT-7B model endpoint. It covers various scenarios including valid API calls, calls that result in errors, and the saving/loading of language model endpoints.

**docs/modules/models/llms/integrations/octoai.ipynb**: This notebook shows how to use LangChain to interact with `OctoAI` [LLM endpoints](https://octoai.cloud/templates)

Please note that to use the newly added OctoAI's services, users should have the octoai python package installed `pip install octoai`, and the environment variables OCTOAI_API_TOKEN set with the API token and ENDPOINT_URL set with endpoint_url. The API token and endpoint URL can be obtained by cloning a template and creating an API token at https://octoai.cloud.

Thanks in advance for your review!

@hwchase17 @agola11

### CODE REVIEW

File: langchain/__init__.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
     HuggingFaceTextGenInference,
     LlamaCpp,
     Modal,
+    OctoAIEndpoint,
```
Comment (hwchase17) [inline langchain/__init__.py:32]: we probably shouldnt add it here

**Review by hwchase17 — APPROVED** [review]

**Review by Bassem-Yacoube — COMMENTED** [review]
Yes, that works. 
Thank you.

**Comment by AI-Bassem** [comment]
I fixed the lint errors but Vercel seems to still be failing. I can't see what the error is with Vercel. Can you please let me know what needs to be fixed?

**Comment by AI-Bassem** [comment]
conflicts resolved. lint and test errors fixed. Vercel deployment now succeeding.

**Pull request merged**